sdk/ai/ai-inference-rest/src/tracingPolicy.ts

```diff
@@ -0,0 +1,252 @@
+        headers: {
+          _headersMap: {
+          },
+        },
+        method: "POST",
+        timeout: 0,
+        multipartBody: undefined,
+        formData: undefined,
+        disableKeepAlive: false,
+        proxySettings: undefined,
+        streamResponseStatusCodes: undefined,
+        withCredentials: false,
+        abortSignal: undefined,
+        tracingOptions: {
+        },
+        onUploadProgress: undefined,
+        onDownloadProgress: undefined,
+        requestId: "e89d3930-257d-4d96-a60b-216e5ecc8040",
+        allowInsecureConnection: false,
+        enableBrowserStreams: true,
+      }
+
+       */
+      if (!request.body)
+        return next(request);
+      const body = JSON.parse(request.body as string);
+
+      const spanAttributes: TracingAttributes = {
+        [TracingAttributesEnum.Operation_Name]: "operation name", ///TODD: don't know how to determine yet
+        [TracingAttributesEnum.Request_Model]: body.model,
+        [TracingAttributesEnum.Request_Frequency_Penalty]: body.frequency_penalty,
+        [TracingAttributesEnum.Request_Max_Tokens]: body.max_tokens,
+        [TracingAttributesEnum.Request_Presence_Penalty]: body.presence_penalty,
+        [TracingAttributesEnum.Request_Stop_Sequences]: body.stop,
+        [TracingAttributesEnum.Request_Temperature]: body.temperature,
+        [TracingAttributesEnum.Request_Top_K]: body.top_k,
+        [TracingAttributesEnum.Request_Top_P]: body.top_p
+      };
+
+      const { span, tracingContext } = tryCreateSpan(tracingClient, request, spanAttributes) ?? {};
+
+      if (!span || !tracingContext) {
+        return next(request);
+      }
+
+      try {
+        const response = await tracingClient.withContext(tracingContext, next, request);
+        tryProcessResponse(span, response);
+        return response;
+      } catch (err: any) {
+        tryProcessError(span, err);
+        throw err;
+      }
+    },
+  };
+}
+
+function tryCreateTracingClient(): TracingClient | undefined {
+  try {
+    return createTracingClient({
+      namespace: "",
+      packageName: "@azure-rest/ai-inference",
+      packageVersion: SDK_VERSION,
+    });
+  } catch (e: unknown) {
+    logger.warning(`Error when creating the TracingClient: ${getErrorMessage(e)}`);
+    return undefined;
+  }
+}
+
+function tryCreateSpan(
+  tracingClient: TracingClient,
+  request: PipelineRequest,
+  spanAttributes: Record<string, unknown>,
+): { span: TracingSpan; tracingContext: TracingContext } | undefined {
+  try {
+    // As per spec, we do not need to differentiate between HTTP and HTTPS in span name.
+    const { span, updatedOptions } = tracingClient.startSpan(
+      "{gen_ai.operation.name} {gen_ai.request.model}", ///TODO: not sure how to retrieve the name yet
```

Comment:
I don't think this naming convention is [per spec](https://opentelemetry.io/docs/specs/semconv/http/http-spans/#name) - @lmolkova what do you think?

Given this is an http tracing span maybe we need to see: 
- name: HTTP {method} which are available from the request
- sanitized URL as part of the attributes

Similar to how we do it in tracing policy in core? https://github.com/Azure/azure-sdk-for-js/blob/main/sdk/core/core-rest-pipeline/src/policies/tracingPolicy.ts

Reply:
I found some properties are included for other spans for other use case.   I am not sure if I should include them or not.   @lmolkova    Do you have a sample result to follow?  Currently my code logs this
![image](https://github.com/user-attachments/assets/33ccfb92-cc2e-42f8-88f5-292e1bdc9455)

Reply:
`az.namespace` should match the corresponding Azure Resource Provider namespace - https://learn.microsoft.com/en-us/azure/azure-resource-manager/management/azure-services-resource-providers#ai-and-machine-learning-resource-providers

I believe it should be `Microsoft.CognitiveServices` ?

Reply:
Set to Microsoft.CognitiveServices now.